### ISSUE TITLE
`glob` gets locked globally

### ISSUE THREAD

**Issue by KAAtheWiseGit** [nushell/nushell#12397]
### Describe the bug

For some reason, `glob` can only run in one process at the same time.  This means that if a time-consuming `glob` is launched in one window, all other globs will have to wait for it to finish.

### How to reproduce

1. Execute some long `glob` in one terminal, i.e. `glob **/*`.
2. Try to execute a second, fast one in another terminal window: `glob .`.  The latter will block until the prior finishes.

### Expected behavior

Commands should be independent of each other.

### Configuration

| key                | value                                                        |
| ------------------ | ------------------------------------------------------------ |
| version            | 0.90.1                                                       |
| branch             |                                                              |
| commit_hash        |                                                              |
| build_os           | linux-x86_64                                                 |
| build_target       | x86_64-alpine-linux-musl                                     |
| rust_version       | rustc 1.76.0 (07dca489a 2024-02-04) (Alpine Linux 1.76.0-r0) |
| cargo_version      | cargo 1.76.0                                                 |
| build_time         | 2024-02-14 02:25:04 +00:00                                   |
| build_rust_channel | release                                                      |
| allocator          | mimalloc                                                     |
| features           | default, sqlite, trash, which, zip                           |
| installed_plugins  |                                                              |

**Comment by fdncred** [comment]
This is not my experience on Windows.

**Comment by devyn** [comment]
I don't really see any reason why this should happen, but I will try anyway. Are you sure it's not just that one glob is slowing the other down?

**Comment by KAAtheWiseGit** [comment]
@fdncred the issue is that `ls` works just fine.  I can do `ls /**/*`, which is even more resource intensive, and othre `glob`'s and `ls`'s will continue to work just fine.  But the moment I execute a long-running `glob`, all other `glob`'s stop.

Which I find weird, because `glob` should be faster than `ls`, since it doesn't check for the metadata.

**Comment by fdncred** [comment]
Agreed. It's weird. I cannot reproduce this behavior.

**Comment by KAAtheWiseGit** [comment]
Update: `glob` is also much slower than `ls`.

```Nushell
~> timeit { ls --full-paths --all * }
180µs 910ns
~> timeit { glob * }
1sec 301ms 658µs 314ns
```

Where both outputs are steams and the resulting files are identical.

**Comment by singh-priyank** [comment]
I also can't reproduce this on Windows system 
In 1st terminal I ran `glob **/*`
And in 2nd one I ran `glob .`
2nd terminal gave the instant output but 1st one was still running

**Comment by KAAtheWiseGit** [comment]
One more point here: `glob` is really slow.  On the current version, it takes one and a half seconds to do complete trivial operations:

```Nushell
~> timeit { glob * }
1sec 421ms 924µs 815ns
~> timeit { ls * }
417µs 516ns
~> timeit { glob . }
1sec 365ms 973µs 548ns
~> timeit { ls --directory . }
39µs 164ns
```

### Version

| key                | value                                                        |
| ------------------ | ------------------------------------------------------------ |
| version            | 0.92.0                                                       |
| branch             |                                                              |
| commit_hash        |                                                              |
| build_os           | linux-x86_64                                                 |
| build_target       | x86_64-alpine-linux-musl                                     |
| rust_version       | rustc 1.77.0 (aedd173a2 2024-03-17) (Alpine Linux 1.77.0-r0) |
| cargo_version      | cargo 1.77.0                                                 |
| build_time         | 2024-04-03 10:33:03 +00:00                                   |
| build_rust_channel | release                                                      |
| allocator          | mimalloc                                                     |
| features           | default, sqlite, trash, which                                |
| installed_plugins  |                                                              |

**Comment by fdncred** [comment]
Agreed. It's slower. I've talked to the author of the wax crate about making it multithreaded but it's not yet. In the meantime, depending on what you're wanting to do, my `nu_plugin_jwalk` is super-fast and has control over threads and sorting. Here's the same tests on my sucky Windows desktop.
```nushell
❯ timeit glob *
765ms 639µs 300ns
❯ timeit ls *
5ms 972µs 500ns
❯ timeit glob .
251ms 465µs 900ns
❯ timeit ls --directory .
1ms 378µs
❯ jwalk c:\Users\username\source\repos\forks\nushell --original --debug --threads 4 --max-depth 1
1.4992ms
```

**Comment by KAAtheWiseGit** [comment]
What I find weird is why would it be slower, given `ls` is very fast.  I know `glob` is more potent with it's expansions, but as long as those complex patterns aren't used, there shouldn't be much of a slowdown.

**Comment by fdncred** [comment]
> What I find weird is why would it be slower, given ls is very fast. I know glob is more potent with it's expansions, but as long as those complex patterns aren't used, there shouldn't be much of a slowdown.

Someone would probably have to profile it and see where it's slowing down. It could be our code or upstream. Not sure.
